### PR TITLE
STR-43 -- DO: Fix Docker-Compose so that Docker works the way it was intended + Laravel Echo

### PR DESCRIPTION
**Pull request by ebm17915**
# Description
Adjusted docker-compose.yml and Dockerfile. Dockerfile installs redis and laravel echo server.
https://blog.mikesir87.io/2018/07/leveraging-multi-stage-builds-single-dockerfile-dev-prod/

I will slim down Dockerfile in next commit.
# Testing
1. Checkout the STR-43 branch.
2.  Make sure Dockerfile+docker-compose.yml is inside iSTART directory
3. Run `docker-compose up -d`
4. Give redis and echo a few minutes to serve up.
5. Run the command `docker logs iSTART_echo` to see if laravel echo is "Server ready!"
# Installation
1. Have Dockerfile+docker-compose.yml file.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,18 +27,11 @@ COPY resources/ /app/resources/
 WORKDIR /app
 RUN yarn install \
     && yarn run prod
-# PHP/Apache 
-FROM csunmetalab/environment:base-20190130
-COPY . /var/www/html
+# PHP/Apache
+FROM csunmetalab/environment:base-20190130 as build
 # Copy Front/Backend packages
+COPY --from=dev /var/www/html /var/www/html
```
Comment (digitalsoba) [inline Dockerfile]: There is no layer called `dev`

**Review by digitalsoba — CHANGES_REQUESTED** [review]
Make the quick change, can't test out Laravel echo but everything else is 👍

**Comment by ruizjorgealt** [comment]
Not accessible through localhost

File: Dockerfile [hunk 2]
```diff
@@ -18,18 +17,54 @@ COPY resources/ /app/resources/
...
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && \
+    apt-get install -qq -y build-essential nodejs yarn \
+    libpq-dev \
+    mysql-client
+RUN yarn global add --prod --no-lockfile laravel-echo-server \
+ && yarn cache clean \
+ && mkdir -p /app/database
+# Install Redis.
+RUN \
```
Comment (digitalsoba) [inline Dockerfile]: Separate Redis into another container in `docker-compose.yml`

**Review by digitalsoba — DISMISSED** [review]
Step 11/22 `Error: ENOENT: no such file or directory, open '/public/css/app.css'` in the build process where `yarn install && yarn run prod`

**Review by luisjg — CHANGES_REQUESTED** [review]
It's missing the php-ldap extension

**Review by luisjg — APPROVED** [review]
👍

**Comment by ruizjorgealt** [comment]
Looks good to me!

**Review by ruizjorgealt — APPROVED** [review]
Works fine!

**Pull request merged**